chcemvediet/apps/accounts/admin.py

```diff
@@ -44,6 +47,10 @@ class ProfileAdmin(admin.ModelAdmin):
                 short_description=u'Undecided E-mails',
                 admin_order_field=u'undecided_emails_count',
                 ),
+            decorate(
+                lambda o: admin_obj_format(o, u'Log in', link=u'login_as'),
+                short_description=u'Login As',
```

Comment:
Tu mame chybu, co som si doteraz nevsimol.  `o` v `ProfileAdmin` je predsa instancia `Profile`, nie `User`. Takze `login_as_view` ako `obj_pk` dostane `Profile.pk` nie `User.pk`. Nikde nemame zarucene, ze user a jeho instancia profilu budu mat rovnake `pk`, takze ak prihlasime usera s pk rovnym `Profile.pk` mozeme prihlasit ineho usera, nez komu dany profil patri.

Aby to fungovalo, "login as" stlpec musime presunut z `ProfileAdmin` do `UserAdmin`. Zatial pouzivame built in `UserAdmin` definovany v djangu. Treba zo s djanga zobrat, rozsirit o `AdminLoginAsAdminMixin`, pridat "Login as" stlpec, a do admin rozhrania zaregistrovat tento nas upraveny `UserAdmin` namiesto povodneho djangoveho `UserAdmin`.

Kedze `User` model nie je sucastou ziadnej nasej app, tak nie som si isty kde nas `UserAdmin` definovat. Skus, ci sa bude dat tento `UserAdmin` definovat priamo v `apps/accounts/admin.py` vedla `ProfileAdmin`. Ak ano, tak to asi nechajme tam. Ak nie, tak pogoogli ako customizovat defaultny djangovy `UserAdmin`

Reply:
@martinmacko47 Este predtym ako sa pustim do riesenia, keby sme upravili model `Profile`, pridanim `primary_key`, bude platit vzdy ze `user.pk == profile.pk`.  Migracia zbehne v poriadku vyskusal som si to na inom projekte, potom len treba prepisat kod na miestach kde sa pouziva `profile.id` a nahradit to `profile.pk` (*popriade `profile.user`)
```
class Profile(FormatMixin, OriginalValuesMixin, models.Model):
    user = models.OneToOneField(User, primary_key=True)
    ...
```

Reply:
@martinmacko47 Dobre uz som to spravil, je to pomerne jednoduche, vid riesenie. Pridal som "Login As" feature aj do `ProfileAdmin` aj do `UserAdmin`. Povedz ak to chces mat iba v jednom.
Ale zaujimalo by ma do buducna ci ma zmysel pouzit `user=OneToOneField(User, primary_key=True)`, ak by sme chceli dosiahnut rovnake `pk` pre oba modely. dakujem.

Reply:
@viliambalaz Finta s `user=OneToOneField(User, primary_key=True)` je velmi cool napad. Pri novych 1to1 tabulkach to tak zacnime robit.

Primary key existujucich tabuliek vsak radsej nemenme. Skusil som si vygenerovat SQL commandy, co to za migraciu spravi, kedze sa meni primary key. Ono to vytvori uplne novu tabulku, obsah povodnej tabulky skopiruje do novej a potom povodnu tabulku drope. Mazat a vytvarat tabulky na zivom projekte je extremne riskantne. Radsej to nerobme.

`AdminLoginAsAdminMixin` nechaj teda iba na `UserAdmin`. Avsak pre praktickost stlpec "Login As" mozes nechat na oboch aj na `UserAdmin` aj na `ProfileAdmin`. Akurat na `ProfileAdmin` linka nebude smerovat na `o` ale na `o.user`. Takze sa pouzije aj view z `UserAdmin`, hoci linka bude z `ProfileAdmin`.

Reply:
:ok: